server/services/teamsSummaryCountService.test.ts

```diff
@@ -0,0 +1,579 @@
+    )
+  })
+
+  it('should orchestrate data flow correctly to build product->component->alert mapping', async () => {
+    const teamSlug = 'team-alpha'
+
+    const mockProducts = [
+      { id: 1, attributes: { name: 'Product A', slug: 'product-a', p_id: 'pa' } },
+      { id: 2, attributes: { name: 'Product B', slug: 'product-b', p_id: 'pb' } },
+    ]
+
+    mockStrapiClient.getTeam.mockResolvedValue({
+      data: [
+        {
+          id: 1,
+          attributes: {
+            t_id: 'team-alpha-id',
+            name: 'Team Alpha',
+            products: { data: mockProducts },
+            createdAt: '2023-01-01T00:00:00Z',
+            createdBy: { data: { attributes: {}, id: 1 } },
+            updatedAt: '2023-01-01T00:00:00Z',
+            updatedBy: { data: { attributes: {}, id: 1 } },
+          },
+        },
+      ],
+    } as unknown as TeamResponse)
+
+    mockStrapiClient.getProduct.mockImplementation((params: ProductParams) => {
+      if (params.productSlug === 'product-a') {
+        return Promise.resolve({
+          data: {
+            id: 1,
+            attributes: {
+              name: 'Product A',
+              slug: 'product-a',
+              p_id: 'pa',
+              components: {
+                data: [
+                  { id: 1, attributes: { name: 'ComponentA1' } },
+                  { id: 2, attributes: { name: 'ComponentA2' } },
+                ],
+              },
+            },
+          },
+        } as unknown as ProductResponse)
+      }
+      return Promise.resolve({
+        data: {
+          id: 2,
+          attributes: {
+            name: 'Product B',
+            slug: 'product-b',
+            p_id: 'pb',
+            components: { data: [{ id: 3, attributes: { name: 'ComponentB1' } }] },
+          },
+        },
+      } as unknown as ProductResponse)
+    })
+
+    const mockAlerts = [
+      { status: { state: 'active' }, labels: { component: 'ComponentA1' } },
+      { status: { state: 'active' }, labels: { component: 'ComponentA1' } },
+      { status: { state: 'active' }, labels: { component: 'ComponentA1' } },
+      { status: { state: 'active' }, labels: { component: 'ComponentB1' } },
+      { status: { state: 'active' }, labels: { component: 'ComponentB1' } },
+    ]
+    mockAlertsService.getAlerts.mockResolvedValue(mockAlerts)
+
+    const result = await service.getTeamAlertSummary(teamSlug)
+
+    expect(mockStrapiClient.getTeam).toHaveBeenCalledWith({ teamSlug })
+    expect(mockStrapiClient.getProduct).toHaveBeenCalledTimes(2)
+    expect(mockAlertsService.getAlerts).toHaveBeenCalled()
+
+    expect(result).toEqual({
+      products: {
+        'Product A': {
+          ComponentA1: 3,
+          ComponentA2: 0,
+        },
+        'Product B': {
+          ComponentB1: 2,
+        },
+      },
+      total: 5,
+    })
+
+    expect(mockLogger.info).toHaveBeenCalledWith(expect.stringContaining('Start for team team-alpha'))
```

Comment:
The logger is not functionality, you shouldn't be testing that the logger is outputting things.

Reply:
Saves having to change tests if you want to change the logging.